src/element-style-observer.js

```diff
@@ -109,7 +109,38 @@ export default class ElementStyleObserver {
 	 * @param {TransitionEvent} event
 	 */
 	async handleEvent (event) {
-		if (!this.properties.has(event.propertyName)) {
+		if (event.type === "animationstart") {
+			let animation = document.getAnimations().find(a => a.effect.target === this.target);
+
+			// Get the observed properties that are being animated by the animation
+			let keyframes = animation.effect.getKeyframes();
+			let properties = new Set(keyframes.flatMap(frame => 
```

Comment:
There's a caveat: We can't get their names from the keyframes if we animate custom properties. 🤷‍♂️

On my local branch, I tried to add all observed custom properties to the `properties` array `properties.push(...this.propertyNames.filter(property => property.startsWith("--")));`.

It worked. But I still wonder, what if it's wrong... 🤔

Reply:
Interestingly, Firefox does return custom properties when getting an animation's keyframes with `getKeyframes().`